### PR TITLE
feat(certificatemanager): add support for additional key algorithms

### PR DESCRIPTION
**Pull request by laurelmay**
This adds support for any key algorithms that are supported for a
certificate request. There are additional supported values; however,
they are not supported for new keys only for imported keys.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-certificatemanager/README.md [hunk 1]
```diff
@@ -156,6 +156,19 @@ new acm.Certificate(this, 'Certificate', {
...
+## Key Algorithms
+
+The [key algorithm](https://docs.aws.amazon.com/acm/latest/userguide/acm-certificate.html#algorithms.title) can be specified
+for each key. Only some values (highlighted in the linked article) are supported for certificate
+requests.
+
+```ts
+new acm.Certificate(this, 'Certificate', {
+  domainName: 'test.example.com',
+  keyAlgorithm: acm.keyAlgorithm.EC_PRIME256V1,
```
Comment (kylelaker) [inline packages/aws-cdk-lib/aws-certificatemanager/README.md:168]:
```suggestion
  keyAlgorithm: acm.KeyAlgorithm.EC_PRIME256V1,
```

File: packages/aws-cdk-lib/aws-certificatemanager/lib/certificate.ts [hunk 2]
```diff
@@ -100,7 +100,41 @@ export interface CertificateProps {
...
+   * RSA is the default key algorithm for ACM certificates. Elliptic Curve Digital Signature
+   * Algorithm (ECDSA) keys are smaller, offering security comparable to RSA keys but with greater
+   * computing efficiency. However, ECDSA is not supported by all network clients. Some AWS
+   * services may require RSA keys, or only support ECDSA keys of a particular size, while others
+   * allow the use of either RSA and ECDSA keys to ensure that compatibility is not broken. Check
+   * the requirements for the AWS service where you plan to deploy your certificate.
+   *
+   * @see https://docs.aws.amazon.com/acm/latest/userguide/acm-certificate.html#algorithms.title
+   *
+   * @default the CloudFormation default (RSA 2048 bit)
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-certificatemanager/lib/certificate.ts:117]:
```suggestion
   * @default - `RSA_2048`

File: packages/aws-cdk-lib/aws-certificatemanager/README.md [hunk 3]
```diff
@@ -156,6 +156,19 @@ new acm.Certificate(this, 'Certificate', {
 });
 ```
 
+## Key Algorithms
+
+The [key algorithm](https://docs.aws.amazon.com/acm/latest/userguide/acm-certificate.html#algorithms.title) can be specified
+for each key. Only some values (highlighted in the linked article) are supported for certificate
+requests.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-certificatemanager/README.md:163]:
```suggestion
To specify the algorithm of the public and private key pair that your certificate uses to encrypt data use the `keyAlgorithm` property.

Algorithms supported for an ACM certificate request include:
 * `RSA_2048`
 * `EC_prime256v1`
 * `EC_secp384r1`
```

File: packages/aws-cdk-lib/aws-certificatemanager/README.md [hunk 4]
```diff
@@ -156,6 +156,19 @@ new acm.Certificate(this, 'Certificate', {
...
+
+```ts
+new acm.Certificate(this, 'Certificate', {
+  domainName: 'test.example.com',
+  keyAlgorithm: acm.KeyAlgorithm.EC_PRIME256V1,
+});
+```
+
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-certificatemanager/README.md:171]:
```suggestion

> Visit [Key algorithms](https://docs.aws.amazon.com/acm/latest/userguide/acm-certificate.html#algorithms.title) for more details.

```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good 👍
I left some notes for documentation adjustments.

**Comment by kylelaker** [comment]
Closing in favor of #28597.

**Pull request closed without merging**